explainability-core/src/main/java/org/kie/trustyai/metrics/language/rouge/ROUGE.java

```diff
@@ -0,0 +1,166 @@
+package org.kie.trustyai.metrics.language.rouge;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.kie.trustyai.metrics.language.AbstractNLPPerformanceMetric;
+import org.kie.trustyai.metrics.language.utils.NGramUtils;
+import org.kie.trustyai.metrics.language.utils.SentenceUtils;
+import org.kie.trustyai.metrics.language.utils.F1Score;
+
+public class ROUGE extends AbstractNLPPerformanceMetric<Double, String>{
+
+    public enum RougeTypes {
```

Comment:
@christinaexyou I think in here we could do away with strings altogether and just use the enum directly. Something like

```java
public enum RougeTypes {
        ROUGE_LSUM,
        ROUGE1,
        ROUGE2,
        ROUGEL;
    }

    private RougeTypes rougeType;

    public ROUGE(RougeTypes rougeType){
        super();
        this.rougeType = rougeType;
    }
```

and

```java
    public Double calculate(String reference, String hypothesis, RougeTypes rougeType) {
        if (rougeType.equals(RougeTypes.ROUGE_LSUM)){
            // Split the reference and hypothesis into sentences
            final String[] referenceSentences = SentenceUtils.splitSentences(reference);
            final String[] hypothesisSentences = SentenceUtils.splitSentences(hypothesis);
            return rougeLsum(referenceSentences, hypothesisSentences);
        }
   ...
```

Reply:
@ruivieira i use the string to differentiate between ROUGE1 / ROUGE2 and avoid writing duplicate code when i have to calculate n-grams

Reply:
This can still be done with enums, for instance

```java
public enum RougeTypes {
    ROUGE_LSUM,
    ROUGE1,
    ROUGE2,
    ROUGEL;

    public int getN() {
        switch (this) {
            case ROUGE1:
                return 1;
            case ROUGE2:
                return 2;
            default:
                throw new UnsupportedOperationException("getN() is not supported for " + this);
        }
    }
}
``` 
and
```java
public class ROUGE extends AbstractNLPPerformanceMetric<Double, String> {

    public Double calculate(String reference, String hypothesis, RougeTypes rougeType) {

  switch (rougeType) {
            case ROUGE_LSUM:
                // ...
            case ROUGEL:
                // ...
            case ROUGE1:
            case ROUGE2:
                final int n = rougeType.getN();
                final Map<String, Integer> referenceNgramCount = NGramUtils.countNgrams(referenceTokens, n);
                final Map<String, Integer> hypothesisNGramCount = NGramUtils.countNgrams(hypothesisTokens, n);
                return rougeN(referenceNgramCount, hypothesisNGramCount);            
            default:
                return 0.0;
        }
    }
```